fab/operations/supervisor.py

```diff
@@ -57,34 +57,49 @@ def _get_celery_queues():
 @roles(ROLES_CELERY)
 @parallel
 def set_celery_supervisorconf():
-
-    conf_files = {
-        'main':                         ['supervisor_celery_main.conf'],
-        'periodic':                     ['supervisor_celery_beat.conf', 'supervisor_celery_periodic.conf'],
-        'sms_queue':                    ['supervisor_celery_sms_queue.conf'],
-        'reminder_queue':               ['supervisor_celery_reminder_queue.conf'],
-        'reminder_rule_queue':          ['supervisor_celery_reminder_rule_queue.conf'],
-        'reminder_case_update_queue':   ['supervisor_celery_reminder_case_update_queue.conf'],
-        'pillow_retry_queue':           ['supervisor_celery_pillow_retry_queue.conf'],
-        'background_queue':             ['supervisor_celery_background_queue.conf'],
-        'saved_exports_queue':          ['supervisor_celery_saved_exports_queue.conf'],
-        'ucr_queue':                    ['supervisor_celery_ucr_queue.conf'],
-        'ucr_indicator_queue':          ['supervisor_celery_ucr_indicator_queue.conf'],
-        'email_queue':                  ['supervisor_celery_email_queue.conf'],
-        'repeat_record_queue':          ['supervisor_celery_repeat_record_queue.conf'],
-        'logistics_reminder_queue':     ['supervisor_celery_logistics_reminder_queue.conf'],
-        'logistics_background_queue':   ['supervisor_celery_logistics_background_queue.conf'],
-        'async_restore_queue':          ['supervisor_celery_async_restore_queue.conf'],
-        'flower':                       ['supervisor_celery_flower.conf'],
-    }
-
     queues = _get_celery_queues()
-    if 'periodic' in queues and env.host != queues['periodic'].get('server_whitelist'):
+
+    if 'celery_periodic' in queues and env.host != queues['celery_periodic'].get('server_whitelist'):
         show_periodic_server_whitelist_message_and_abort(env)
+
     _rebuild_supervisor_conf_file('make_supervisor_conf', 'celery_bash_runner.sh')
+
     for queue, params in queues.items():
-        for config_file in conf_files[queue]:
-            _rebuild_supervisor_conf_file('make_supervisor_conf', config_file, {'celery_params': params})
+        if queue == 'flower':
+            _rebuild_supervisor_conf_file(
+                'make_supervisor_conf',
+                'supervisor_celery_flower.conf',
+                {'celery_params': params}
+            )
+            continue
+
+        pooling = params.get('pooling', 'prefork')
+        max_tasks_per_child = params.get('max_tasks_per_child', 50)
+        num_workers = params.get('num_workers', 1)
+
+        params.update({
+            'queue': queue,
+            'pooling': pooling,
+            'max_tasks_per_child': max_tasks_per_child,
+        })
+
+        for worker_num in range(num_workers):
+            params.update({
+                'worker_num': worker_num,
+            })
+
+            _rebuild_supervisor_conf_file(
```

Comment:
yeah it creates one config file per worker as it was doing before. i don't really feel strongly on which is better - i don't think there's many pros/cons to either approach, besides the fact that it was easier to do it this way for celery because of how we start and stop the workers, so i think will be easier to maintain.

keeping pillows the way they are is probably also fine if it's easier to maintain that way for that process group.